zcash_client_sqlite/src/wallet.rs

```diff
@@ -727,13 +741,37 @@ pub(crate) fn get_wallet_summary<P: consensus::Parameters>(
                 balances.add_unshielded_value(value)?;
             }
         }
+        drop(transparent_trace);
     }
 
+    let next_sapling_subtree_index = {
+        // This will return a runtime error if we call `get_wallet_summary` from two
+        // threads at the same time, as transactions cannot nest.
+        let tx = conn.unchecked_transaction()?;
```

Comment:
Why not have this whole method take a `rusqlite::Transaction` rather than a `rusqlite::Connection`?